Internal: Support adding static attributes to atomic element [ED-20586]

Omerisra6 opened the pull request:
<!--start_gitstream_placeholder-->
### ✨ PR Description
Purpose: Add support for static attributes in atomic elements through a new define_initial_attributes() method in the element base class.
Main changes:
- Added define_initial_attributes() method to Atomic_Element_Base for defining static attributes
- Updated render attribute handling to merge initial attributes with dynamic ones
- Created new Atomic_Tab_Panel element with role="tabpanel" attribute
- Fixed Atomic_Tabs_List class name and renamed from Atomic_Tab_List

_Generated by LinearB AI and added by gitStream._
<sub>AI-generated content may contain inaccuracies. Please verify before using. **[We'd love your feedback!](mailto:product@linearb.io)** 🚀</sub>
<!--end_gitstream_placeholder-->

Omerisra6 committed:
Support adding static attributes to atomic element [ED-20586]

Omerisra6 committed:
wip

elementorbot committed:
Merge branch 'main' into ED-20586

IshayMaya commented on modules/atomic-widgets/elements/atomic-element-base.php:
```diff
@@ -108,7 +113,7 @@ protected function print_html_tag() {
 	protected function print_custom_attributes() {
 		$settings = $this->get_atomic_settings();
-		$attributes = $settings['attributes'];
+		$attributes = $settings['attributes'] ?? [];
```
according to line 119 it can also be string?

IshayMaya commented on modules/atomic-widgets/elements/atomic-tabs/atomic-tab-link.php:
```diff
@@ -79,10 +79,28 @@ protected function define_base_styles(): array {
 
+	protected function define_static_attributes() {
+		return [
+			'role' => 'button',
```
looking at some examples online (and our current tabs) it should probably be a button with role="tab"

IshayMaya commented on modules/atomic-widgets/elements/atomic-tabs/atomic-tab-link.php:
```diff
@@ -99,14 +118,4 @@ protected function add_render_attributes() {
 
 		$this->add_render_attribute( '_wrapper', $attributes );
 	}
```
need to add the heading back in define_default_children

IshayMaya commented on modules/atomic-widgets/elements/atomic-tabs/atomic-tab-panel.php:
```diff
@@ -0,0 +1,125 @@
+
+	protected function define_default_children() {
+		return [
+			Atomic_Tabs_Content::generate()
```
should be the other direction no?
contents renders panel (tabpanel is also the name of the actual role)

IshayMaya commented on modules/atomic-widgets/elements/atomic-tabs/atomic-tab-panel.php:
```diff
@@ -0,0 +1,125 @@
+		parent::add_render_attributes();
+		$settings = $this->get_atomic_settings();
+		$base_style_class = $this->get_base_styles_dictionary()[ static::BASE_STYLE_KEY ];
+		$static_attributes = $this->define_static_attributes();
```
role="tabpanel"

elementorbot committed:
Merge branch 'main' into ED-20586

elementorbot committed:
Merge branch 'main' into ED-20586

elementorbot committed:
Merge branch 'main' into ED-20586

Omerisra6 committed:
Refactors attribute handling for Atomic Widgets

Renames 'static_attributes' to 'initial_attributes' for better clarity,
reflecting their purpose as default values. This change improves the
consistency of attribute handling within Atomic Widgets.

Ensures initial attributes are merged correctly with other attributes
during element rendering.

Updates the default children for Atomic Tabs content.

Omerisra6 committed:
Merge remote-tracking branch 'Omerisra6/ED-20586' into ED-20586

elementorbot committed:
Merge branch 'main' into ED-20586

Omerisra6 committed:
wip

elementorbot committed:
Merge branch 'main' into ED-20586

Omerisra6 committed:
wip

elementorbot committed:
Merge branch 'main' into ED-20586

Omerisra6 committed:
Merge remote-tracking branch 'Omerisra6/ED-20586' into ED-20586

elementorbot committed:
Merge branch 'main' into ED-20586

elementorbot committed:
Merge branch 'main' into ED-20586

elementorbot committed:
Merge branch 'main' into ED-20586

elementorbot committed:
Merge branch 'main' into ED-20586
